main.py

```diff
@@ -246,6 +253,48 @@ def generate_json():
 
     return {"room_days": room_days_info}
 
+@app.route('/json/<day>/<room>')
+@catch_error
+def generate_json_individual(day, room):
+    # Check for access level
+    _, level = access(require_name=False)
+    if level < 1:
+        access_error()
+
+    unreviewed = expect(request, "unreviewed", optional=True)
+
+    room_days_info = []
+
+    room_days = room_days_query_individual(day, room).all()
+    for room_day in room_days:
```

Comment:
Can put everything from this line to just before the return in a function that accepts `room_days` and returns essentially `room_days_info`, and have both `generate_json_individual` and `generate_json` call it. Not required.